net/tcp/tcp_conn.c

```diff
@@ -1039,6 +1039,10 @@ FAR struct tcp_conn_s *tcp_alloc_accept(FAR struct net_driver_s *dev,
       conn->tcpstateflags = TCP_SYN_RCVD;
 
       tcp_initsequence(conn->sndseq);
+#if !defined(CONFIG_NET_TCP_WRITE_BUFFERS)
```

Comment:
```suggestion
#if !defined(CONFIG_NET_TCP_WRITE_BUFFERS) || \
    defined(CONFIG_NET_SENDFILE)
```

Reply:
Why to CONFIG_NET_SENDFILE? I'm testing here these scenarios:
1) no TCP_WRITE_BUFFERS
2) TCP_WRITE_BUFFERS but NO SENDFILE
3) TCP_WRITE_BUFFERS and SENDFILE

In all these cases it works fine without add defined(CONFIG_NET_SENDFILE)